src/c4/runtime_check.cc

```diff
@@ -0,0 +1,36 @@
+//----------------------------------*-C++-*----------------------------------//
+/*!
+ * \file   ds++/runtime_check.cc
+ * \author Kent Grimmett Budge
+ * \date   Wed Mar 28 07:58:48 2018
+ * \brief  Member definitions of class runtime_check
+ * \note   Copyright (C) 2018 Los Alamos National Security, LLC.
+ *         All rights reserved. */
+//---------------------------------------------------------------------------//
+
+#include <exception>
+
+#include "C4_Functions.hh"
+#include "ds++/DracoStrings.hh"
+
+namespace rtt_dsxx {
+
+//-------------------------------------------------------------------------//
+void runtime_check(bool condition, char const *message) noexcept(false) {
+  Require(message != nullptr);
+
+  unsigned sum = !condition;
+  rtt_c4::global_sum(sum);
+  if (sum != 0) // some processors failed the condition
+  {
+    throw std::runtime_error("runtime error detected on " +
+                             rtt_dsxx::to_string(sum) + " processor(s): " +
+                             message);
+  }
+}
```

Comment:
Is there any value it making this look more like a Design-by-Contract assertion?  It would be easy to add CPP macros that wrap this function.  Something like `Parallel_Require( a>0 )` or `pInsist( b>0, "b must be positive )`.  The `global_sum` could be costly and might need to be turned off for fully optimized builds.  If you go this route, consider renaming these files to be variants of `Assert.hh` -- something like `Parallel_Assert.hh` or `MPI_Assert.hh`.

Reply:
I did not originally write this with DBC in mind. I was thinking of actual run time errors, such as a file opening or write that fails, or bad user input, or a calculation that is well posed that runs off the rails (temperatures go negative or a matrix becomes numerically singular.) We want these things in the production builds, not just for debugging.  However, it's not unreasonable to think about a DBC wrapper for synchronous tests that we don't necessarily want turned on in production code.

Reply:
I think your UseCase is reasonable.  The function could be called directly (as you intended) or via a `mpiInsist` CPP macro (these are never disabled) but has the advantage of having a familiar name.